Quartz/Engine/Source/Core/Utilities/FileIO.cpp

```diff
@@ -42,3 +42,8 @@ std::string FileIO::readAllFile(const std::string& filepath)
 	return fileString;
 }
 
+std::string FileIO::getDirname(const std::string& filename)
+{
+	std::size_t p = filename.find_last_of("\\/");
```

Comment:
How about using the filesystem stuff?
Like
```cpp
inline std::string getDirectoryFromPath(const std::string& path)
        {
            return (std::filesystem::path(path)).parent_path().string();
        }

        inline std::string getFilenameFromPath(const std::string& path)
        {
            return (std::filesystem::path(path)).filename().string();
        }
```

Reply:
None of that is in C++11 AFAIK, it'll only be mainstream in C++20

Reply:
[std::filesystem::path](https://en.cppreference.com/w/cpp/filesystem/path)  is C++17, either way still not C++11, is this project to remain C++11?

Reply:
Right, yoyre probably right, I just thought everything in filesystem was experimental, but either way, I think we're sticking with C++11 until most things progress to 17. We should probably have a proper discussion on discord about this, the pros and cons to each one.

Reply:
Oh crap, I forgot that it was only C++17 :(